README.md

```diff
@@ -894,239 +912,207 @@ How to test actions with scenario:
 ## Actions without parameters
 
 ```python
-from scenario import Context, State, ActionOutput
+import scenario
+
 from charm import MyCharm
 
 
 def test_backup_action():
-    ctx = Context(MyCharm)
+    ctx = scenario.Context(MyCharm)
 
-    # If you didn't declare do_backup in the charm's `actions.yaml`, 
-    # the `ConsistencyChecker` will slap you on the wrist and refuse to proceed.
-    out: ActionOutput = ctx.run_action("do_backup_action", State())
+    # If you didn't declare do_backup in the charm's `actions.yaml` or `charmcraft.yaml`, 
+    # the consistency checker will slap you on the wrist and refuse to proceed.
+    out: ActionOutput = ctx.run_action(MyCharm.on.do_backup_action, scenario.State())
 
-    # you can assert action results, logs, failure using the ActionOutput interface
+    # You can assert action results, logs, and failure using the ActionOutput interface:
     assert out.logs == ['baz', 'qux']
-    
-    if out.success:
-      # if the action did not fail, we can read the results:
-      assert out.results == {'foo': 'bar'}
 
-    else:
-      # if the action fails, we can read a failure message
-      assert out.failure == 'boo-hoo'
+    # We can read the results - these might be partial if the action failed.
+    assert out.results == {'foo': 'bar'}
+
+    if not out.success:
```

Comment:
There's that whole discussion of whether this should be an exception. Does that apply here?